test/jest/foregroundTimeTracker.spec.ts

```diff
@@ -0,0 +1,153 @@
+import ForegroundTimeTracker from '../../src/foregroundTimeTracker';
+
+describe('ForegroundTimeTracker', () => {
+    let foregroundTimeTracker: ForegroundTimeTracker;
+    const apiKey = 'test-key';
+    const mockStorageKey = `mp-time-${apiKey}`;
+
+    beforeEach(() => {
+        // Although in Jest, document.hidden should be false by default, we force it to be this way
+        // to ensure the page is in the foreground
+        Object.defineProperty(document, 'hidden', { value: false, configurable: true });
+        jest.useFakeTimers();
+        localStorage.clear();
+    });
+
+    afterEach(() => {
+        jest.clearAllTimers();
+        jest.restoreAllMocks();
+    });
+
+    it('should initialize with correct localStorage key', () => {
```

Comment:
From an organizational point, I would like to see the tests organized by function being tested. It would make it easier to find things in the future and better organize/collect reused setup and teardown steps.

Reply:
this is a good call that i just completely overlooked.